cli/src/ui.rs

```diff
@@ -258,12 +257,32 @@ pub enum PaginationChoice {
     Auto,
 }
 
+enum PagerConfig {
+    Disabled,
+    Builtin,
+    External(CommandNameAndArgs),
+}
+
+impl PagerConfig {
+    fn from_config(config: &StackedConfig) -> Result<PagerConfig, ConfigGetError> {
+        let PaginationChoice::Auto = config.get("ui.paginate")? else {
+            return Ok(PagerConfig::Disabled);
```

Comment:
nit: `if matches!(.., PaginationChoice::Auto)` looks slightly better.

Reply:
OK, it's now

```
        if !matches!(config.get("ui.paginate")?, PaginationChoice::Auto) {
```

Reply:
Oh, I should say `if matches!(.., Never)`. If you don't have any preference, I would use `if never => disabled`.

Reply:
Sure, SG.